src/shared/sam/localLambdaRunner.ts

```diff
@@ -363,20 +363,20 @@ export async function invokeLambdaFunction(
     return config
 }
 
-function stopApi(config: SamLaunchRequestArgs) {
-    if (!samStartApi) {
+function stopApi(config: SamLaunchRequestArgs, proc: ChildProcess | undefined) {
+    if (!proc) {
         // This is a bug, should never happen.
         getLogger().error('SAM: unknown debug session: %s', config.name)
         return
     }
 
     try {
-        getLogger().verbose('SAM: sending SIGHUP to sam process: pid %d', samStartApi.pid())
-        samStartApi.stop(true, 'SIGHUP')
+        getLogger().verbose('SAM: sending SIGHUP to sam process: pid %d', proc.pid())
+        proc.stop(true, 'SIGHUP')
     } catch (e) {
-        getLogger().warn('SAM: failed to stop sam process: pid %d: %O', samStartApi.pid(), e as Error)
+        getLogger().warn('SAM: failed to stop sam process: pid %d: %O', proc.pid(), e as Error)
```

Comment:
Should we consider this a failure? Should we attempt a `SIGKILL`?

Reply:
`proc.stop(true, ...)` does SIGKILL implicitly if the process does not exit after some time. This try-catch is just defensive :)